ArmRobot/Assets/Scripts/Robot/RobotVisionCapture.cs

```diff
@@ -0,0 +1,111 @@
+                                                    yAltitudeTable, robotMaxReach - minimumDistanceWithoutObjects);
+            }
+            else{
+                sphere.transform.position = new Vector3(robotMaxReach - minimumDistanceWithoutObjects - minimumDistanceBetweenObjects*(i-maxNumberObjectsPerRow+1), 
+                                                    yAltitudeTable, robotMaxReach - minimumDistanceWithoutObjects - minimumDistanceBetweenObjects);
+            }
+            
+            sphere.transform.localScale = new Vector3(scaleObject, scaleObject, scaleObject);
+            sphere.AddComponent<ColorRandomizer>();
+            sphere.AddComponent<RandomizerPositionObject>();
+            sphere.tag = "Sphere";
+            sphere.name = "Sphere_" + i;
+ }
+    }
+
+
+    private void Update()
+    {
+        VisionDataCollector visionDataCollector = GetComponent<VisionDataCollector>();
+        string imageName = visionDataCollector.NextImageName();
+        Vector3 relativeCubePosition = cube.transform.position - robot.transform.position;
+        RobotVisionDataPoint dataPoint = new RobotVisionDataPoint(relativeCubePosition, imageName);
+
+        bool didCapture = visionDataCollector.CaptureIfNecessary(imageName, dataPoint);
+
+        if (didCapture)
+        {
+            Reset();
+        }
+    }
+
+    // HELPERS
+
+    private void Reset()
+    {
+        // we create a list that will contains all the objects we have already moved 
+        List<GameObject> listOfAlreadyMovedObjects = new List<GameObject>();
+
+        // move cube
+        TablePositionRandomizer tablePositionRandomizer_cube = cube.GetComponent<TablePositionRandomizer>();
+        tablePositionRandomizer_cube.Move();
+
+        listOfAlreadyMovedObjects.Add(cube);
+
+        // move other objects 
+        // first we create an array of all the objects in the scene except the cube 
+
+        GameObject[] arrayOfSpheres= GameObject.FindGameObjectsWithTag("Sphere");
+        List<GameObject> listOfObjects = new List<GameObject>();
+        listOfObjects.AddRange(arrayOfSpheres); // we need to do that because we want to work with list and not array and arrayOfSpheres is an array 
+        listOfObjects.Add(cylinder);
+
+        // then we will iterate through the arrayOfObjects and move them one by one 
+        foreach (GameObject gameobject in listOfObjects) {
+            RandomizerPositionObject tablePositionRandomizerObject = gameobject.GetComponent<RandomizerPositionObject>();
+            tablePositionRandomizerObject.Move(listOfAlreadyMovedObjects);
+            listOfAlreadyMovedObjects.Add(gameobject); // we add the gameobject to the already moved objects 
+
+        }
+
+        // move robot 
+        RobotController robotController = robot.GetComponent<RobotController>();
+        float[] rotation = {Random.value * 10, Random.value * 10, Random.value * 10, Random.value * 10, Random.value * 10, Random.value * 10};
```

Comment:
It would be cleaner to make `MoveRobot` a separate method, where you can pass in the max movement in degrees.

Reply:
Only moving the joints 10 degrees in just the positive direction? I agree we need to be cautious, because we don't want to make the robot try to go through itself or the table, but we could at least do both positive and negative rotation changes.